.github/workflows/update-kaveri-data.yml

```diff
@@ -0,0 +1,157 @@
+name: Automated Kaveri Data Update
+
+on:
+  schedule:
+    # Run weekly full scrape on Sunday at 3 AM UTC  
+    - cron: '0 3 * * 0'
+  
+  workflow_dispatch:
+    inputs:
+      force_full_scrape:
+        description: 'Force full scrape regardless of changes'
+        required: false
+        default: false
+        type: boolean
+
+permissions:
+  contents: write
+  actions: read
+  issues: write
+
+jobs:
+  update-kaveri-data:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+    
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.9'
+    
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install requests
+    
+    - name: Update credentials in config
+      env:
+        KAVERI_AUTH: ${{ secrets.KAVERI_AUTH_TOKEN }}
+        KAVERI_COOKIE: ${{ secrets.KAVERI_COOKIE }}
+      run: |
+        cd kaveri
+        cat > update_creds.py << 'EOF'
+        import json
+        import os
+        from datetime import datetime
+
+        # Load config
+        with open('config.json', 'r') as f:
+            config = json.load(f)
+
+        # Update credentials from secrets
+        if os.getenv('KAVERI_AUTH') and os.getenv('KAVERI_COOKIE'):
+            config['credentials']['authorization'] = os.getenv('KAVERI_AUTH')
+            config['credentials']['cookie'] = os.getenv('KAVERI_COOKIE')
+            config['credentials']['last_updated'] = datetime.now().isoformat()
+            
+            with open('config.json', 'w') as f:
+                json.dump(config, f, indent=2)
+            print('✓ Credentials updated from secrets')
+        else:
+            print('⚠️ No credentials found in secrets')
+        EOF
+        python3 update_creds.py
+    
+    - name: Run data scraper
+      id: scraper
+      run: |
+        cd kaveri
+        python3 automated_scraper.py
+        
+        # Check if files were updated and provide summary
+        if git diff --quiet; then
```

Comment:
Let's stick to using one language, (Python, for example), and not switch between shell and Python syntax. If the goal is to leverage unix tools and composition with pipes, Python can execute shell commands with `subprocess.run()` can be used.